src/lib/services/repository-access/RepositorySearchService.ts

```diff
@@ -13,173 +11,318 @@ import {
     wrapErrorCode,
     wrapSuccess,
 } from 'lib/util/apiResponseWrapper/apiResponseWrapper';
+import { IAssetAdministrationShellRepositoryApi, ISubmodelRepositoryApi } from 'lib/api/basyx-v3/apiInterface';
 
-export type RepoSearchResult = {
-    aas: AssetAdministrationShell;
+export type RepoSearchResult<T> = {
+    searchResult: T;
     location: string;
 };
 
-export interface NullableMultipleDataSourceSetupParameters {
-    shellsSavedInTheRepositories?: INullableAasRepositoryEntries[] | null;
-    submodelsSavedInTheRepository?: Submodel[] | null;
-    log?: Log | null;
-}
+const noDefaultAasRepository = <T>() =>
+    wrapErrorCode<T>(ApiResultStatus.INTERNAL_SERVER_ERROR, 'No default AAS repository configured');
+const noDefaultSubmodelRepository = <T>() =>
+    wrapErrorCode<T>(ApiResultStatus.INTERNAL_SERVER_ERROR, 'No default Submodel repository configured');
 
 export class RepositorySearchService {
```

Comment:
One can now clearly distinguish between searchInDefault, getFirstFromAllRepos and getAllFromAllRepos.
We may want to extract the aas and the submodel repository search logic though